File: android/app/src/main/java/com/cartoncloud/model/WeatherInfo.java
package com.cartoncloud.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Main model object to store the weather information.
 */
public class WeatherInfo implements Serializable {
    @SerializedName("id")
    @Expose
    private long id;
    @SerializedName("weather_state_name")
    @Expose
    private String weatherStateName;
    @SerializedName("weather_state_abbr")
    @Expose
    private String weatherStateAbbr;
    @SerializedName("wind_direction_compass")
    @Expose
    private String windDirectionCompass;
    @SerializedName("created")
    @Expose
    private String created;
    @SerializedName("applicable_date")
    @Expose
    private String applicableDate;
    @SerializedName("min_temp")
    @Expose
    private double minTemp;
    @SerializedName("max_temp")
    @Expose
    private double maxTemp;
    @SerializedName("the_temp")
    @Expose
    private double theTemp;
    @SerializedName("wind_speed")
    @Expose
    private double windSpeed;
    @SerializedName("wind_direction")
    @Expose
    private double windDirection;
    @SerializedName("air_pressure")
    @Expose
    private double airPressure;
    @SerializedName("humidity")
    @Expose
    private int humidity;
    @SerializedName("visibility")
    @Expose
    private double visibility;
    @SerializedName("predictability")
    @Expose
    private int predictability;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getWeatherStateName() {
        return weatherStateName;
    }

    public void setWeatherStateName(String weatherStateName) {
        this.weatherStateName = weatherStateName;
    }

    public String getWeatherStateAbbr() {
        return weatherStateAbbr;
    }

    public void setWeatherStateAbbr(String weatherStateAbbr) {
        this.weatherStateAbbr = weatherStateAbbr;
    }

    public String getWindDirectionCompass() {
        return windDirectionCompass;
    }

    public void setWindDirectionCompass(String windDirectionCompass) {
        this.windDirectionCompass = windDirectionCompass;
    }

    public String getCreated() {
        return created;
    }

    public void setCreated(String created) {
        this.created = created;
    }

    public String getApplicableDate() {
        return applicableDate;
    }

    public void setApplicableDate(String applicableDate) {
        this.applicableDate = applicableDate;
    }

    public double getMinTemp() {
        return minTemp;
    }

    public void setMinTemp(double minTemp) {
        this.minTemp = minTemp;
    }

    public double getMaxTemp() {
        return maxTemp;
    }

    public void setMaxTemp(double maxTemp) {
        this.maxTemp = maxTemp;
    }

    public double getTheTemp() {
        return theTemp;
    }

    public void setTheTemp(double theTemp) {
        this.theTemp = theTemp;
    }

    public double getWindSpeed() {
        return windSpeed;
    }

    public void setWindSpeed(double windSpeed) {
        this.windSpeed = windSpeed;
    }

    public double getWindDirection() {
        return windDirection;
    }

    public void setWindDirection(double windDirection) {
        this.windDirection = windDirection;
    }

    public double getAirPressure() {
        return airPressure;
    }

    public void setAirPressure(double airPressure) {
        this.airPressure = airPressure;
    }

    public int getHumidity() {
        return humidity;
    }

    public void setHumidity(int humidity) {
        this.humidity = humidity;
    }

    public double getVisibility() {
        return visibility;
    }

    public void setVisibility(double visibility) {
        this.visibility = visibility;
    }

    public int getPredictability() {
        return predictability;
    }

    public void setPredictability(int predictability) {
        this.predictability = predictability;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof WeatherInfo)) return false;

        WeatherInfo that = (WeatherInfo) o;

        if (getId() != that.getId()) return false;
        if (Double.compare(that.getMinTemp(), getMinTemp()) != 0) return false;
        if (Double.compare(that.getMaxTemp(), getMaxTemp()) != 0) return false;
        if (Double.compare(that.getTheTemp(), getTheTemp()) != 0) return false;
        if (Double.compare(that.getWindSpeed(), getWindSpeed()) != 0) return false;
        if (Double.compare(that.getWindDirection(), getWindDirection()) != 0) return false;
        if (Double.compare(that.getAirPressure(), getAirPressure()) != 0) return false;
        if (getHumidity() != that.getHumidity()) return false;
        if (Double.compare(that.getVisibility(), getVisibility()) != 0) return false;
        if (getPredictability() != that.getPredictability()) return false;
        if (getWeatherStateName() != null ? !getWeatherStateName().equals(that.getWeatherStateName()) : that.getWeatherStateName() != null)
            return false;
        if (getWeatherStateAbbr() != null ? !getWeatherStateAbbr().equals(that.getWeatherStateAbbr()) : that.getWeatherStateAbbr() != null)
            return false;
        if (getWindDirectionCompass() != null ? !getWindDirectionCompass().equals(that.getWindDirectionCompass()) : that.getWindDirectionCompass() != null)
            return false;
        if (getCreated() != null ? !getCreated().equals(that.getCreated()) : that.getCreated() != null)
            return false;
        return getApplicableDate() != null ? getApplicableDate().equals(that.getApplicableDate()) : that.getApplicableDate() == null;
    }

    @Override
    public int hashCode() {
        int result;
        long temp;
        result = (int) (getId() ^ (getId() >>> 32));
        result = 31 * result + (getWeatherStateName() != null ? getWeatherStateName().hashCode() : 0);
        result = 31 * result + (getWeatherStateAbbr() != null ? getWeatherStateAbbr().hashCode() : 0);
        result = 31 * result + (getWindDirectionCompass() != null ? getWindDirectionCompass().hashCode() : 0);
        result = 31 * result + (getCreated() != null ? getCreated().hashCode() : 0);
        result = 31 * result + (getApplicableDate() != null ? getApplicableDate().hashCode() : 0);
        temp = Double.doubleToLongBits(getMinTemp());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(getMaxTemp());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(getTheTemp());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(getWindSpeed());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(getWindDirection());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(getAirPressure());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        result = 31 * result + getHumidity();
        temp = Double.doubleToLongBits(getVisibility());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        result = 31 * result + getPredictability();
        return result;
    }
}

File: android/app/src/main/java/com/cartoncloud/model/WeatherInfo.kt
package com.cartoncloud.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Main model object to store the weather information.
 */
class WeatherInfo {
    @SerializedName("id")
    @Expose
    var id: Int? = null
    @SerializedName("weather_state_name")
    @Expose
    var weatherStateName: String? = null
    @SerializedName("weather_state_abbr")
    @Expose
    var weatherStateAbbr: String? = null
    @SerializedName("wind_direction_compass")
    @Expose
    var windDirectionCompass: String? = null
    @SerializedName("created")
    @Expose
    var created: String? = null
    @SerializedName("applicable_date")
    @Expose
    var applicableDate: String? = null
    @SerializedName("min_temp")
    @Expose
    var minTemp: Any? = null
    @SerializedName("max_temp")
    @Expose
    var maxTemp: Any? = null
    @SerializedName("the_temp")
    @Expose
    var theTemp: Double? = null
    @SerializedName("wind_speed")
    @Expose
    var windSpeed: Double? = null
    @SerializedName("wind_direction")
    @Expose
    var windDirection: Double? = null
    @SerializedName("air_pressure")
    @Expose
    var airPressure: Double? = null
    @SerializedName("humidity")
    @Expose
    var humidity: Int? = null
    @SerializedName("visibility")
    @Expose
    var visibility: Double? = null
    @SerializedName("predictability")
    @Expose
    var predictability: Int? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is WeatherInfo) return false

        if (id != other.id) return false
        if (weatherStateName != other.weatherStateName) return false
        if (weatherStateAbbr != other.weatherStateAbbr) return false
        if (windDirectionCompass != other.windDirectionCompass) return false
        if (created != other.created) return false
        if (applicableDate != other.applicableDate) return false
        if (minTemp != other.minTemp) return false
        if (maxTemp != other.maxTemp) return false
        if (theTemp != other.theTemp) return false
        if (windSpeed != other.windSpeed) return false
        if (windDirection != other.windDirection) return false
        if (airPressure != other.airPressure) return false
        if (humidity != other.humidity) return false
        if (visibility != other.visibility) return false
        if (predictability != other.predictability) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id ?: 0
        result = 31 * result + (weatherStateName?.hashCode() ?: 0)
        result = 31 * result + (weatherStateAbbr?.hashCode() ?: 0)
        result = 31 * result + (windDirectionCompass?.hashCode() ?: 0)
        result = 31 * result + (created?.hashCode() ?: 0)
        result = 31 * result + (applicableDate?.hashCode() ?: 0)
        result = 31 * result + (minTemp?.hashCode() ?: 0)
        result = 31 * result + (maxTemp?.hashCode() ?: 0)
        result = 31 * result + (theTemp?.hashCode() ?: 0)
        result = 31 * result + (windSpeed?.hashCode() ?: 0)
        result = 31 * result + (windDirection?.hashCode() ?: 0)
        result = 31 * result + (airPressure?.hashCode() ?: 0)
        result = 31 * result + (humidity ?: 0)
        result = 31 * result + (visibility?.hashCode() ?: 0)
        result = 31 * result + (predictability ?: 0)
        return result
    }
}